src/middlewared/middlewared/plugins/chart_releases_linux/validation.py

```diff
@@ -136,3 +163,36 @@ async def validate_container_image(self, verrors, value, question, schema_name,
         else:
             if not digest:
                 verrors.add(schema_name, f'Unable to retrieve {tag!r} container image tag details.')
+
+    @private
+    async def validate_custom_host_path(self, verrors, path, question, schema_name, release_data):
+        if not path:
+            return
+
+        is_cluster = path.startswith(FuseConfig.FUSE_PATH_SUBST.value)
+        if is_cluster:
+            verrors.add(schema_name, f"Fuse mounted paths are not allowed: {path}.")
+
+        p = Path(path)
+        if not p.is_absolute():
+            verrors.add(schema_name, f"Must be an absolute path: {path}.")
+
+        if os.path.realpath(path).startswith("/root/.ssh"):
+            verrors.add(schema_name, f"Changes to permissions under /root/.ssh are not permitted: {path}.")
+
+        if not os.path.realpath(path).startswith("/mnt/"):
+            verrors.add(
+                schema_name,
+                f"Path should not be beneath the /mnt directory: {path}."
+            )
+        elif len(p.resolve().parents) == 2:
+            verrors.add(schema_name, f"The specified path is a ZFS pool mountpoint: {path}")
+        elif await self.middleware.call('pool.dataset.path_in_locked_datasets', path):
+            verrors.add(schema_name, f"Dataset is locked at path: {path}.")
+
+        apps_dataset = (await self.middleware.call('kubernetes.config'))['dataset']
+        if apps_dataset and os.path.realpath(path).startswith(f'/mnt/{apps_dataset}'):
```

Comment:
I think let's be generic here and do what we do in `check_path_resides_within_volume` validator so that wherever we see `ix-applications` under pool mountpoint, we don't allow it to be consumed